src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallerFactory.cs

```diff
@@ -20,7 +22,32 @@ internal class WorkloadInstallerFactory
             string dotnetDir = null, 
             PackageSourceLocation packageSourceLocation = null)
         {
+            var installType = GetWorkloadInstallType(sdkFeatureBand);
+
+            if (installType == InstallType.Msi && OperatingSystem.IsWindows())
+            {
+                return new NetSdkMsiInstaller();
+            }
+
             return new NetSdkManagedInstaller(reporter, sdkFeatureBand, workloadResolver, nugetPackageDownloader, verbosity: verbosity, dotnetDir: dotnetDir, packageSourceLocation: packageSourceLocation);
         }
+
+        /// <summary>
+        /// Determines the <see cref="InstallType"/> associated with a specific SDK version.
+        /// </summary>
+        /// <param name="sdkFeatureBand">The SDK version to check.</param>
+        /// <returns>The <see cref="InstallType"/> associated with the SDK.</returns>
+        public static InstallType GetWorkloadInstallType(SdkFeatureBand sdkFeatureBand)
+        {
+            string installerTypePath = Path.Combine(Path.GetDirectoryName(Environment.ProcessPath), "metadata",
+                "workloads", $"{sdkFeatureBand}", "installertype");
```

Comment:
nit: `$"{sdkFeatureBand}"` is the same as `sdkFeatureBand`?

Reply:
Is it? The interpolation does an implicit ToString(), otherwise I need to do sdkFeatureBand.ToString() to convert it I think

Reply:
Oh gotcha, I'm not sure.